### PR TITLE
[ML] Updated parameter name to be idle_time_before_shutdown_minutes

### PR DESCRIPTION
**Pull request by nthandeMS**
# Description

Updated `idle_time_before_shutdown` to `idle_time_before_shutdown_minutes` to make the property easier to use and fit the intended spec. 
  - Now uses integer minutes as input instead of ISO time format.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by singankit — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py [hunk 1]
```diff
@@ -122,9 +122,9 @@ class ComputeInstance(Compute):
     :type schedules: Optional[ComputeSchedules], optional
     :param identity:  The identity configuration, identities that are associated with the compute cluster.
     :type identity: IdentityConfiguration, optional
-    :param idle_time_before_shutdown: Stops compute instance after user defined period of
-        inactivity. Time is defined in ISO8601 format. Minimum is 15 min, maximum is 3 days.
-    :type idle_time_before_shutdown: Optional[str], optional
+    :param idle_time_before_shutdown_minutes: Stops compute instance after a user defined period of
```
[the remark this answers is not in the document]
Comment (luigiw) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:125]: @singankit do we need to do that for experimental classes?

[the remark this answers is not in the document]
Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:125]: @singankit Could you weigh in on this? Would like to get this in soon.

[the remark this answers is not in the document]
Comment (singankit) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:126]: For experimental it should probably be okay.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py [hunk 2]
```diff
@@ -141,7 +142,7 @@ def __init__(
         ssh_settings: Optional[ComputeInstanceSshSettings] = None,
         schedules: Optional[ComputeSchedules] = None,
         identity: IdentityConfiguration = None,
-        idle_time_before_shutdown: Optional[str] = None,
+        idle_time_before_shutdown_minutes: Optional[str] = None,
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:145]: Should type be Optional[int] here?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py [hunk 3]
```diff
@@ -230,10 +235,14 @@ def _to_rest_object(self) -> ComputeResource:
             subnet=subnet_resource,
             ssh_settings=ssh_settings,
             personal_compute_instance_settings=personal_compute_instance_settings,
-            idle_time_before_shutdown=self.idle_time_before_shutdown,
+            idle_time_before_shutdown=f"PT{self.idle_time_before_shutdown_minutes}M",
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:238]: Nit: this works for numbers bigger than 60 right?

**Review by luigiw — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py (hunk 2 above)
Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py]: Good catch.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py (hunk 3 above)
Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_compute/compute_instance.py:238]: Yep -- up to PT4320M, which is 72 hours.

File: sdk/ml/azure-ai-ml/tests/compute/unittests/test_compute_entity.py [hunk 4]
```diff
@@ -140,6 +140,7 @@ def test_compute_instance_schedules_from_yaml(self):
         assert compute_instance2.schedules.compute_start_stop[1].trigger.frequency == "week"
         assert compute_instance2.schedules.compute_start_stop[1].trigger.interval == 1
         assert compute_instance2.schedules.compute_start_stop[1].trigger.schedule is not None
+        assert compute_instance2.idle_time_before_shutdown_minutes == 15
```
Comment (singankit) [inline sdk/ml/azure-ai-ml/tests/compute/unittests/test_compute_entity.py:143]: Can we please add a test which validated the conversion to rest object and if this property gets translated as expected by rest api ?

**Review by singankit — APPROVED** [review]
Please add the suggested test.

File: sdk/ml/azure-ai-ml/tests/compute/unittests/test_compute_entity.py (hunk 4 above)
Comment (nthandeMS) [reply sdk/ml/azure-ai-ml/tests/compute/unittests/test_compute_entity.py]: Added a line into this test to validate the number was converted to the expected ISO time format.

**Pull request merged**